### PR TITLE
Some doctests for basedocs

### PR DESCRIPTION
**Pull request by kshyatt**
I added some xrefs and backticks too.

### CODE REVIEW

**Comment by kshyatt** [comment]
Travis fail is unrelated `libgit2`. AV fail seems like the continuing problem there. Can someone review this?

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -1181,6 +1199,12 @@ end
     Symbol(x...) -> Symbol
 
 Create a `Symbol` by concatenating the string representations of the arguments together.
+
+# Examples
+```jldoctest
+julia> Symbol("my", "name")
```
Comment (fredrikekre) [inline base/docs/basedocs.jl]: Just a suggestion; To highlight the "string representations of the arguments" part better perhaps not use two `String`s?

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -1218,7 +1242,27 @@ getfield
     setfield!(value, name::Symbol, x)
 
 Assign `x` to a named field in `value` of composite type. The syntax `a.b = c` calls
-`setfield!(a, :b, c)`.
+`setfield!(a, :b, c)`. `value` must be mutable.
+
+# Examples
+```jldoctest
+julia> a = 1//2
```
Comment (fredrikekre) [inline base/docs/basedocs.jl]: I would switch the examples to show a working example first. Also, perhaps make a custom `struct` too? It is not obvious that `Rational` has a `den` field.

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -1301,27 +1345,68 @@ Array{T,N}(dims)
...
 -(x, y)
 
 """
     *(x, y...)
 
 Multiplication operator. `x*y*z*...` calls this function with all arguments, i.e. `*(x, y, z, ...)`.
+
+# Examples
+```jldoctest
+julia> 2*7*8
```
Comment (fredrikekre) [inline base/docs/basedocs.jl]: Nitpicking, but perhaps nicer with some space: `2 * 7 * 8`

File: base/docs/basedocs.jl (hunk 2 above)
Comment (kshyatt) [reply base/docs/basedocs.jl]: I was using `Rational` to make a connection with the `getfield` example directly above. I can switch to the field we want to access being `num` as in that example.

**Comment by kshyatt** [comment]
Hopefully I got everything @fredrikekre <3. I added another example to `getfield` as well to show the `a.b` stuff.

**Review by KristofferC — APPROVED** [review]

**Review by fredrikekre — APPROVED** [review]

**Comment by kshyatt** [comment]
OSX fail is unrelated:

```
InitError(mod=:LinAlg, error=ErrorException("error compiling __init__: error compiling check: could not load library "libopenblas"
dlopen(libopenblas.dylib, 1): image not found"))
```

**Pull request merged**